modules/storage-api/src/main/java/org/apache/ignite/internal/storage/InvokeClosure.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage;
+
+import org.jetbrains.annotations.Nullable;
+
+/** */
+public interface InvokeClosure {
+    /**
+     * @param row Old row or {@code null} if old row not found.
+     */
+    void call(@Nullable DataRow row);
+
+    /**
+     * @return New row for {@link OperationType#WRITE} operation.
+     */
+    DataRow newRow();
```

Comment:
It seems like this method is only needed for write operations. Should it really be a part of the Closure's API?

Reply:
How else would you provide a new row?